pkg/client/codec.go

```diff
@@ -0,0 +1,20 @@
+package client
+
+import (
+	"errors"
+	"net/url"
+
+	"k8s.io/apimachinery/pkg/conversion/queryparams"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type parameterCodec struct{}
```

Comment:
Could we add a `var _ runtime.ParamterCodec = parameterCodec` to make this easier to understand? Also, please add some godocs to it clarifying how exactly it differs from the `ParamterCodec` obtained via the scheme

Reply:
I think a more correct approach to take is setting the serializer on the restclient when we fetch it from the cache. I think that will allow us to remove this code and streamline further. 

If that doesn't work out, i'll add some comments that this is for no-conversion serialization of create/delete/etc params.

For comparison sake, I looked at the dynamic client go client which forces everything to metav1. So it's technically correct right now due to metav1 being the only option, but setting the serializer is the more future-proof option.

Reply:
added a comment and renamed for clarity